cmd/bucket-replication.go

```diff
@@ -2283,3 +2287,94 @@ func saveResyncStatus(ctx context.Context, bucket string, brs BucketReplicationR
 	configFile := path.Join(bucketMetaPrefix, bucket, replicationDir, resyncFileName)
 	return saveConfig(ctx, objectAPI, configFile, buf)
 }
+
+// getReplicationDiff returns unreplicated objects in a channel
+func getReplicationDiff(ctx context.Context, objAPI ObjectLayer, bucket string, diffCh chan madmin.DiffInfo, opts madmin.ReplDiffOpts) error {
+	objInfoCh := make(chan ObjectInfo)
+	if err := objAPI.Walk(ctx, bucket, opts.Prefix, objInfoCh, ObjectOptions{}); err != nil {
+		logger.LogIf(ctx, err)
+		return err
+	}
+	cfg, err := getReplicationConfig(ctx, bucket)
+	if err != nil {
+		logger.LogIf(ctx, err)
+		return err
+	}
+	tgts, err := globalBucketTargetSys.ListBucketTargets(ctx, bucket)
+	if err != nil {
+		logger.LogIf(ctx, err)
+		return err
+	}
+	rcfg := replicationConfig{
+		Config:  cfg,
+		remotes: tgts,
+	}
+	go func() {
+		defer close(diffCh)
+		for {
+			select {
+			case obj, ok := <-objInfoCh:
+				if !ok {
+					return
+				}
+				roi := getHealReplicateObjectInfo(obj, rcfg)
+				switch roi.ReplicationStatus {
+				case replication.Completed, replication.Replica:
+					if !opts.Verbose {
+						continue
+					}
+					fallthrough
+				default:
+					// ignore pre-existing objects that don't satisfy replication rule(s)
+					if roi.ReplicationStatus.Empty() && !roi.ExistingObjResync.mustResync() {
+						continue
+					}
+					// Ignore object prefixes which are excluded
+					// from versioning via the MinIO bucket versioning extension.
+					if globalBucketVersioningSys.PrefixSuspended(bucket, obj.Name) {
+						continue
+					}
```

Comment:
shouldn't this be done right after you get the `obj, ok` ?

Reply:
incorporated your feedback in the PR.